keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,929 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+Encryption at rest has been a stable Kubernetes for a long time. Every now and
+then there had been cases where, be it by improper handling or external system
+failures, the cluster encryption got into a broken state.
+
+If a single object of a resource type cannot be decrypted, listing resources of that
+type in a path prefix containing the object always fails, even if the rest of
+the resource instances is accessible.
+
+Currently, removing a resource that causes such failures is not possible.
+A cluster administrator must access etcd directly and remove the malformed data manually.
+
+This KEP proposes a way to identify resources that fail to decrypt, and introduces
+a new delete option to completely ignore any storage checks in order to be able
+to delete such a failing resource by using just Kubernetes API.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- provide a way to identify objects that failed decryption
```

Comment:
this seems like it should be broadened to persisted resources that cannot be decoded for any reason